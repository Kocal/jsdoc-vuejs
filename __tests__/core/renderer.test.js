const fixtureProps = require('./__fixtures__/props')
const fixtureData = require('./__fixtures__/data')
const fixtureComputed = require('./__fixtures__/computed')
const render = require('../../lib/core/renderer')

xdescribe('core.renderer', () => {
  beforeAll(() => {
    console.log = jest.fn()
  })

  afterAll(() => {
    console.log.mockRestore()
  })

  ;['default', 'docstrap', 'minami', 'pixi', 'tui'].forEach(renderer => {
    test(`core.renderer.${renderer}`, () => {
      expect(render(renderer, [], [], [])).toMatchSnapshot()
      expect(render(renderer, fixtureProps, [], [])).toMatchSnapshot()
      expect(render(renderer, [], fixtureData, [])).toMatchSnapshot()
      expect(render(renderer, [], [], fixtureComputed)).toMatchSnapshot()
      expect(render(renderer, fixtureProps, fixtureData, fixtureComputed)).toMatchSnapshot()
    })
  })
})
