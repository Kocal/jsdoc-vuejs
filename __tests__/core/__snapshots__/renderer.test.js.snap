// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`core.renderer core.renderer.default 1`] = `"</p></div></div><div class=\\"container-overview\\"><div><p>"`;

exports[`core.renderer core.renderer.default 2`] = `"[object Object],[object Object]</p></div></div><div class=\\"container-overview\\"><div><p>"`;

exports[`core.renderer core.renderer.default 3`] = `"</p></div></div><h3 class=\\"subsection-title\\">Props</h3>[{\\"name\\":\\"counter\\",\\"type\\":{\\"names\\":[\\"Number\\"]},\\"description\\":\\"Current counter's value\\"}]<div class=\\"container-overview\\"><div><p>"`;

exports[`core.renderer core.renderer.default 4`] = `"</p></div></div><h3 class=\\"subsection-title\\">Data</h3>[{\\"name\\":\\"message\\",\\"type\\":{\\"names\\":[\\"String\\"]},\\"description\\":\\"A message\\"}]<div class=\\"container-overview\\"><div><p>"`;

exports[`core.renderer core.renderer.default 5`] = `"[object Object],[object Object]</p></div></div><h3 class=\\"subsection-title\\">Props</h3>[{\\"name\\":\\"counter\\",\\"type\\":{\\"names\\":[\\"Number\\"]},\\"description\\":\\"Current counter's value\\"}]<h3 class=\\"subsection-title\\">Data</h3>[{\\"name\\":\\"message\\",\\"type\\":{\\"names\\":[\\"String\\"]},\\"description\\":\\"A message\\"}]<div class=\\"container-overview\\"><div><p>"`;

exports[`core.renderer core.renderer.docstrap 1`] = `"</p></div></div><div class=\\"container-overview\\"><div><p>"`;

exports[`core.renderer core.renderer.docstrap 2`] = `"[object Object],[object Object]</p></div></div><div class=\\"container-overview\\"><div><p>"`;

exports[`core.renderer core.renderer.docstrap 3`] = `"</p></div></div><h3 class=\\"subsection-title\\">Props</h3>[{\\"name\\":\\"counter\\",\\"type\\":{\\"names\\":[\\"Number\\"]},\\"description\\":\\"Current counter's value\\"}]<div class=\\"container-overview\\"><div><p>"`;

exports[`core.renderer core.renderer.docstrap 4`] = `"</p></div></div><h3 class=\\"subsection-title\\">Data</h3>[{\\"name\\":\\"message\\",\\"type\\":{\\"names\\":[\\"String\\"]},\\"description\\":\\"A message\\"}]<div class=\\"container-overview\\"><div><p>"`;

exports[`core.renderer core.renderer.docstrap 5`] = `"[object Object],[object Object]</p></div></div><h3 class=\\"subsection-title\\">Props</h3>[{\\"name\\":\\"counter\\",\\"type\\":{\\"names\\":[\\"Number\\"]},\\"description\\":\\"Current counter's value\\"}]<h3 class=\\"subsection-title\\">Data</h3>[{\\"name\\":\\"message\\",\\"type\\":{\\"names\\":[\\"String\\"]},\\"description\\":\\"A message\\"}]<div class=\\"container-overview\\"><div><p>"`;

exports[`core.renderer core.renderer.minami 1`] = `"</p></div></div><div class=\\"container-overview\\"><div><p>"`;

exports[`core.renderer core.renderer.minami 2`] = `"[object Object],[object Object]</p></div></div><div class=\\"container-overview\\"><div><p>"`;

exports[`core.renderer core.renderer.minami 3`] = `"</p></div></div><h3 class=\\"subsection-title\\">Props</h3>[{\\"name\\":\\"counter\\",\\"type\\":{\\"names\\":[\\"Number\\"]},\\"description\\":\\"Current counter's value\\"}]<div class=\\"container-overview\\"><div><p>"`;

exports[`core.renderer core.renderer.minami 4`] = `"</p></div></div><h3 class=\\"subsection-title\\">Data</h3>[{\\"name\\":\\"message\\",\\"type\\":{\\"names\\":[\\"String\\"]},\\"description\\":\\"A message\\"}]<div class=\\"container-overview\\"><div><p>"`;

exports[`core.renderer core.renderer.minami 5`] = `"[object Object],[object Object]</p></div></div><h3 class=\\"subsection-title\\">Props</h3>[{\\"name\\":\\"counter\\",\\"type\\":{\\"names\\":[\\"Number\\"]},\\"description\\":\\"Current counter's value\\"}]<h3 class=\\"subsection-title\\">Data</h3>[{\\"name\\":\\"message\\",\\"type\\":{\\"names\\":[\\"String\\"]},\\"description\\":\\"A message\\"}]<div class=\\"container-overview\\"><div><p>"`;

exports[`core.renderer core.renderer.pixi 1`] = `"</p></div></div><div class=\\"container-overview\\"><div><p>"`;

exports[`core.renderer core.renderer.pixi 2`] = `"[object Object],[object Object]</p></div></div><div class=\\"container-overview\\"><div><p>"`;

exports[`core.renderer core.renderer.pixi 3`] = `"</p></div></div><h3 class=\\"subsection-title\\">Props</h3>[{\\"name\\":\\"counter\\",\\"type\\":{\\"names\\":[\\"Number\\"]},\\"description\\":\\"Current counter's value\\"}]<div class=\\"container-overview\\"><div><p>"`;

exports[`core.renderer core.renderer.pixi 4`] = `"</p></div></div><h3 class=\\"subsection-title\\">Data</h3>[{\\"name\\":\\"message\\",\\"type\\":{\\"names\\":[\\"String\\"]},\\"description\\":\\"A message\\"}]<div class=\\"container-overview\\"><div><p>"`;

exports[`core.renderer core.renderer.pixi 5`] = `"[object Object],[object Object]</p></div></div><h3 class=\\"subsection-title\\">Props</h3>[{\\"name\\":\\"counter\\",\\"type\\":{\\"names\\":[\\"Number\\"]},\\"description\\":\\"Current counter's value\\"}]<h3 class=\\"subsection-title\\">Data</h3>[{\\"name\\":\\"message\\",\\"type\\":{\\"names\\":[\\"String\\"]},\\"description\\":\\"A message\\"}]<div class=\\"container-overview\\"><div><p>"`;

exports[`core.renderer core.renderer.tui 1`] = `"</p></div></div><div class=\\"container-overview\\"><div><p>"`;

exports[`core.renderer core.renderer.tui 2`] = `"[object Object],[object Object]</p></div></div><div class=\\"container-overview\\"><div><p>"`;

exports[`core.renderer core.renderer.tui 3`] = `"</p></div></div><h3 class=\\"subsection-title\\">Props</h3>[{\\"name\\":\\"counter\\",\\"type\\":{\\"names\\":[\\"Number\\"]},\\"description\\":\\"Current counter's value\\"}]<div class=\\"container-overview\\"><div><p>"`;

exports[`core.renderer core.renderer.tui 4`] = `"</p></div></div><h3 class=\\"subsection-title\\">Data</h3>[{\\"name\\":\\"message\\",\\"type\\":{\\"names\\":[\\"String\\"]},\\"description\\":\\"A message\\"}]<div class=\\"container-overview\\"><div><p>"`;

exports[`core.renderer core.renderer.tui 5`] = `"[object Object],[object Object]</p></div></div><h3 class=\\"subsection-title\\">Props</h3>[{\\"name\\":\\"counter\\",\\"type\\":{\\"names\\":[\\"Number\\"]},\\"description\\":\\"Current counter's value\\"}]<h3 class=\\"subsection-title\\">Data</h3>[{\\"name\\":\\"message\\",\\"type\\":{\\"names\\":[\\"String\\"]},\\"description\\":\\"A message\\"}]<div class=\\"container-overview\\"><div><p>"`;
